runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/streaming/ComputationStateCacheTest.java

```diff
@@ -0,0 +1,249 @@
+    MapTask mapTask = new MapTask().setStageName("stageName").setSystemName("systemName");
+    Map<String, String> stateNameMap = ImmutableMap.of("stateName1", "stateName");
+    ComputationConfig computationConfig = ComputationConfig.create(mapTask, null, stateNameMap);
+    when(configFetcher.fetchConfig(eq(computationId))).thenReturn(Optional.of(computationConfig));
+    computationStateCache.get(computationId);
+    assertThat(computationStateCache.getGlobalUsernameToStateFamilyNameMap())
+        .containsExactlyEntriesIn(stateNameMap);
+  }
+
+  @Test
+  public void testGetIfPresent_computationStateCached() {
+    String computationId = "computationId";
+    MapTask mapTask = new MapTask().setStageName("stageName").setSystemName("systemName");
+    Map<String, String> userTransformToStateFamilyName =
+        ImmutableMap.of("userTransformName", "stateFamilyName");
+    ComputationConfig computationConfig =
+        ComputationConfig.create(mapTask, userTransformToStateFamilyName, ImmutableMap.of());
+    when(configFetcher.fetchConfig(eq(computationId))).thenReturn(Optional.of(computationConfig));
+    computationStateCache.get(computationId);
+    Optional<ComputationState> computationState = computationStateCache.getIfPresent(computationId);
+    assertTrue(computationState.isPresent());
+    assertThat(computationState.get().getComputationId()).isEqualTo(computationId);
+    assertThat(computationState.get().getMapTask()).isEqualTo(mapTask);
+    assertThat(computationState.get().getTransformUserNameToStateFamily())
+        .isEqualTo(userTransformToStateFamilyName);
+  }
+
+  @Test
+  public void testGetIfPresent_computationStateNotCached() {
+    Optional<ComputationState> computationState =
+        computationStateCache.getIfPresent("computationId");
+    assertFalse(computationState.isPresent());
+    verifyNoInteractions(configFetcher);
+  }
+
+  @Test
+  public void testGetAllComputations() {
+    String computationId1 = "computationId1";
+    String computationId2 = "computationId2";
+    MapTask mapTask = new MapTask().setStageName("stageName").setSystemName("systemName");
+    Map<String, String> userTransformToStateFamilyName =
+        ImmutableMap.of("userTransformName", "stateFamilyName");
+    ComputationConfig computationConfig =
+        ComputationConfig.create(mapTask, userTransformToStateFamilyName, ImmutableMap.of());
+    when(configFetcher.fetchConfig(eq(computationId1))).thenReturn(Optional.of(computationConfig));
+    when(configFetcher.fetchConfig(eq(computationId2))).thenReturn(Optional.of(computationConfig));
+
+    computationStateCache.get(computationId1);
+    computationStateCache.get(computationId2);
+    Set<String> computationIds = ImmutableSet.of(computationId1, computationId2);
+    computationStateCache
+        .getAllPresentComputations()
+        .forEach(
+            computationState -> {
+              assertThat(computationIds).contains(computationState.getComputationId());
```

Comment:
As is this would pass if getAllPresentComputations returned nothing.

verify that it contains both computation ids